session/auth/authprovider/authprovider.go

```diff
@@ -89,6 +96,26 @@ func (ap *authProvider) FetchToken(ctx context.Context, req *auth.FetchTokenRequ
 		Secret:   creds.Secret,
 	}
 
+	var httpClient = http.DefaultClient
+	if tc, err := ap.tlsConfig(req.Host); err == nil && tc != nil {
+		transport := &http.Transport{
```

Comment:
Looks like we already import https://github.com/hashicorp/go-cleanhttp so I think we can reuse that one.